jobsdb/jobsdb.go

```diff
@@ -617,52 +643,52 @@ func (jd *HandleT) workersAndAuxSetup(ownerType OwnerType, tablePrefix string, r
 	config.RegisterBoolConfigVariable(true, &jd.enableWriterQueue, true, enableWriterQueueKeys...)
 	enableReaderQueueKeys := []string{"JobsDB." + jd.tablePrefix + "." + "enableReaderQueue", "JobsDB." + "enableReaderQueue"}
 	config.RegisterBoolConfigVariable(true, &jd.enableReaderQueue, true, enableReaderQueueKeys...)
-	jd.storeChannel = make(chan writeJob)
-	jd.storeWithRetryChannel = make(chan writeJob)
-	jd.updateJobStatusChannel = make(chan writeJob)
-	jd.deleteExecutingChannel = make(chan writeJob)
+	// jd.storeChannel = make(chan writeJob)
+	// jd.storeWithRetryChannel = make(chan writeJob)
+	// jd.updateJobStatusChannel = make(chan writeJob)
+	// jd.deleteExecutingChannel = make(chan writeJob)
+	jd.writeChannel = make(chan writeJob)
 	jd.readChannel = make(chan readJob)
 
 	maxWritersKeys := []string{"JobsDB." + jd.tablePrefix + "." + "maxWriters", "JobsDB." + "maxWriters"}
 	config.RegisterIntConfigVariable(1, &jd.maxWriters, false, 1, maxWritersKeys...)
 	maxReadersKeys := []string{"JobsDB." + jd.tablePrefix + "." + "maxReaders", "JobsDB." + "maxReaders"}
 	config.RegisterIntConfigVariable(3, &jd.maxReaders, false, 1, maxReadersKeys...)
-	jd.initDBWriters()
-	jd.initDBReaders()
 }
 
-func (jd *HandleT) setUpForOwnerType(ownerType OwnerType, clearAll bool) {
+func (jd *HandleT) setUpForOwnerType(ctx context.Context, ownerType OwnerType, clearAll bool) {
 	switch ownerType {
 	case Read:
-		jd.readerSetup()
+		jd.readerSetup(ctx)
 	case Write:
 		jd.setupDatabaseTables(clearAll)
-		jd.writerSetup()
+		jd.writerSetup(ctx)
 	case ReadWrite:
 		jd.setupDatabaseTables(clearAll)
-		jd.readerWriterSetup()
+		jd.readerWriterSetup(ctx)
 	}
 }
 
-func (jd *HandleT) startBackupDSLoop() {
+func (jd *HandleT) startBackupDSLoop(ctx context.Context) {
 	var err error
 	if jd.BackupSettings.BackupEnabled {
 		jd.jobsFileUploader, err = jd.getFileUploader()
 		jd.assertError(err)
-		rruntime.Go(func() {
-			jd.backupDSLoop()
-		})
+		jd.backgroundGroup.Go(misc.WithBugsnag(func() error {
+			jd.backupDSLoop(ctx)
+			return nil
+		}))
 	}
-
 }
 
-func (jd *HandleT) startMigrateDSLoop() {
-	rruntime.Go(func() {
-		jd.migrateDSLoop()
-	})
+func (jd *HandleT) startMigrateDSLoop(ctx context.Context) {
+	jd.backgroundGroup.Go(misc.WithBugsnag(func() error {
+		jd.migrateDSLoop(jd.backgroundCtx)
+		return nil
+	}))
 }
 
-func (jd *HandleT) readerSetup() {
+func (jd *HandleT) readerSetup(ctx context.Context) {
```

Comment:
here ctx and jd.backgroundCtx is the same .. Can we eliminate backgroundCtx if it isn't required